zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/metric/ZkClientMonitor.java

```diff
@@ -96,6 +96,12 @@ public static ObjectName getObjectName(String monitorType, String monitorKey,
             (monitorKey + (monitorInstanceName == null ? "" : "." + monitorInstanceName)));
   }
 
+  public void setAndInitZkEventThreadMonitor(ZkEventThread zkEventThread) {
+    if (_zkEventThreadMetric == null) {
+      _zkEventThreadMetric = new ZkThreadMetric(zkEventThread);
```

Comment:
nit, throw Exception if the _zkEventThreadMetric has already been set or the monitor object has already been registered. Otherwise, we may see an unexpected monitor value in the final result.

Reply:
BTW, you can check for registration status by checking _objectName with a synchronized lock on the monitor object.

Reply:
What type of exception would be good here?

Reply:
Let me put a return value as false and let the calling layer log error